internal/providers/terraform/aws/api_gateway_rest_api.go

```diff
@@ -0,0 +1,159 @@
+package aws
+
+import (
+    "github.com/infracost/infracost/internal/schema"
+    "github.com/shopspring/decimal"
+)
+
+func GetApiGatewayRestApiRegistryItem() *schema.RegistryItem {
+    return &schema.RegistryItem{
+        Name:  "aws_api_gateway_rest_api",
+        RFunc: NewApiGatewayRestApi,
+    }
+}
+
+func NewApiGatewayRestApi(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+    region := d.Get("region").String()
+
+    var apiTierRequests = map[string]decimal.Decimal{
+        "tierOne":   decimal.Zero,
+        "tierTwo":   decimal.Zero,
+        "tierThree": decimal.Zero,
+        "tierFour":  decimal.Zero,
+    }
+
+    monthlyRequests := decimal.Zero
+
+    if u != nil && u.Get("monthly_requests.0.value").Exists() {
+        monthlyRequests = decimal.NewFromInt(u.Get("monthly_requests.0.value").Int())
+    }
+
+    apiRequestQuantities := calculateApiRequests(monthlyRequests, apiTierRequests)
+
+    tierOne   := apiRequestQuantities["tierOne"]
+    tierTwo   := apiRequestQuantities["tierTwo"]
+    tierThree := apiRequestQuantities["tierThree"]
+    tierFour  := apiRequestQuantities["tierFour"]
+
+    return &schema.Resource{
+        Name: d.Address,
+        CostComponents: []*schema.CostComponent{
```

Comment:
Can we only show the first tier if monthlyRequests is zero?
Also, can we only show the applicable tiers where the monthlyRequest is not zero? i.e. if monthlyRequest=500M, tiers one and two are shown. 
That would keep the output short.

Reply:
```
 NAME                          MONTHLY QTY  UNIT          PRICE    HOURLY COST  MONTHLY COST  

  aws_api_gateway_rest_api.api                                                                 
  ├─ Requests (first 333M)              333  1M requests    3.5000       1.5966    1,165.5000  
  └─ Requests (next 667M)              66.7  10M requests  28.0000       2.5584    1,867.6000  
  Total                                                                  4.1549    3,033.1000  
                                                                                               
  aws_api_gateway_stage.cache                                                                  
  └─ Cache memory 0.5                   730  hours          0.0200       0.0200       14.6000  
  Total                                                                  0.0200       14.6000  
                                                                                               
  OVERALL TOTAL (USD)                                                    4.1749    3,047.7000  

```